mypy/checker.py

```diff
@@ -4725,11 +4725,11 @@ def visit_if_stmt(self, s: IfStmt) -> None:
 
                 # XXX Issue a warning if condition is always False?
                 with self.binder.frame_context(can_skip=True, fall_through=2):
-                    self.push_type_map(if_map)
+                    self.push_type_map(if_map, False)
```

Comment:
Also consider using `from_assignment=False` here and elsewhere in this file (when just literal `True`/`False` is passed).